doc/source/cluster/kubernetes/user-guides/upgrade-guide.md

```diff
@@ -0,0 +1,39 @@
+(kuberay-upgrade-guide)=
+
+# KubeRay upgrade guide
+
+## KubeRay / Ray compatibility
+
+KubeRay CI tests the nightly KubeRay against the three most recent minor releases of Ray, as well as against the nightly Ray.
+For example, if the latest Ray release is 2.7.0, KubeRay CI tests the nightly KubeRay against Ray 2.7.0, 2.6.0, 2.5.0, and the nightly Ray.
+
+* KubeRay v0.6.0: Supports all Ray versions > Ray 2.0.0
+* KubeRay v1.0.0: Supports all Ray versions > Ray 2.0.0
+* KubeRay v1.1.0: Planned to release with Ray 2.10.0. It supports Ray 2.8.0 and later
+
+The preceding compatibility plan is closely tied to the KubeRay CRD versioning plan.
+
+## CRD versioning
+
+Typically, while new fields are added to the KubeRay CRD in each release, KubeRay doesn't bump the CRD version for every release.
+
+* KubeRay v0.6.0 and older: CRD v1alpha1
+* KubeRay v1.0.0: CRD v1alpha1 and v1
+* KubeRay v1.1.0: CRD v1
+
+If you want to understand the reasoning behind the CRD versioning plan, see [ray-project/ray#40357](https://github.com/ray-project/ray/pull/40357) for more details.
+
+## Upgrade KubeRay
+
+Upgrading KubeRay version is the best strategy if you have any issues with KubeRay.
+
+* Because a lot of users are unable to install Kubernetes webhooks due to their security policies, KubeRay doesn't provide a webhook for the CRD upgrade.
+* If you plan to upgrade to KubeRay v1.0.0 or later, you may need to upgrade the `apiVersion` in your custom resource YAML files from `ray.io/v1alpha1` to `ray.io/v1`.
+* Based on [the Helm documentation](https://helm.sh/docs/chart_best_practices/custom_resource_definitions/#some-caveats-and-explanations), there is no support at this time for upgrading or deleting CRDs using Helm.
+  * If you want to install the latest KubeRay release's CRD, you may need to delete the old CRD first.
```

Comment:
This will be the case for almost all users, right? Ideally in this doc we can include the error message users see if they forget to do this, so that the error is searchable

Reply:
Update: https://github.com/ray-project/ray/pull/40555/commits/aac8d667981270695e95689e8a3924b121807a20